src/util/stream.h

```diff
@@ -0,0 +1,70 @@
+/*
+Minetest
+Copyright (C) 2022 Minetest Authors
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#pragma once
+
+#include <iostream>
+#include <string>
+#include <functional>
+
+template<int BufferLength, typename Emitter = std::function<void(const std::string &)> >
+class StringStreamBuffer : public std::streambuf {
+public:
+	StringStreamBuffer(Emitter emitter) : m_emitter(emitter) {
+		buffer_index = 0;
+	}
+
+	int overflow(int c) {
+		push_back(c);
+		return c;
+	}
```

Comment:
docs say that the argument to this may be eof https://en.cppreference.com/w/cpp/io/basic_streambuf/overflow
I'm not sure whether this method is even needed.

Reply:
Apologies for the delay, I had made this comment 8 hours ago but didn't realize it ended up "Pending" and not being shown:

The stream never gets closed, so I would never expect EOF to be sent. But if it was, it would be interpreted as byte 0xFF.

Every stream byte goes through overflow(). The push_back method is technically not needed, it could be combined into overflow. I copied this from the prior implementation which had that distinction. Would you prefer I combine them?

Reply:
Eh yeah makes sense.